games/devtest/mods/unittests/init.lua

```diff
@@ -119,10 +119,17 @@ local function wait_for_map(pos, callback)
 	check()
 end
 
+local function printf(fmt, ...)
+	print(fmt:format(...))
+end
+
 -- This runs in a coroutine so it uses await()
 function unittests.run_all()
 	local counters = { time = 0, total = 0, passed = 0 }
 
+	local seed = core.get_us_time()
+	math.randomseed(seed)
```

Comment:
I fear that in practice everything is too non-deterministic (e.g. some code calls `math.random` in between tests) for this to be really useful.
Conceptually I think we should avoid using random numbers in tests - at least for input data.

Reply:
> I fear that in practice everything is too non-deterministic (e.g. some code calls `math.random` in between tests) for this to be really useful.

Fair point. What if we add a `random = true` field to the test def table and if this is set, seed random for each test right before it runs individually? Alternatively we could also pass a `PcgRandom` instance to such tests.

> I think we should avoid using random numbers in tests - at least for input data.

I don't think so. Random numbers are a very powerful tool to test that some assumptions (such as the one unexpectedly violated here) hold for many different configurations with little code.

Reply:
> What if we add a random = true field to the test def table and if this is set, seed random for each test right before it runs individually?

that sounds good.